packages/walletconnect/src/index.ts

```diff
@@ -10,61 +9,44 @@ export const URI_AVAILABLE = 'URI_AVAILABLE'
 
 type MockWalletConnectProvider = WalletConnectProvider & EventEmitter
 
-function parseChainId(chainId: string | number) {
-  return typeof chainId === 'string' ? Number.parseInt(chainId) : chainId
-}
-
-type WalletConnectOptions = Omit<IWCEthRpcConnectionOptions, 'rpc' | 'infuraId' | 'chainId'> & {
-  rpc: { [chainId: number]: string | string[] }
+// @todo document why `rpcMap` type is changed
+type WalletConnectOptions = Omit<Parameters<typeof WalletConnectProvider.init>[0], 'rpcMap'> & {
+  rpcMap?: { [chainId: number]: string | string[] }
 }
 
 /**
  * @param options - Options to pass to `@walletconnect/ethereum-provider`
- * @param defaultChainId - The chainId to connect to in activate if one is not provided.
  * @param timeout - Timeout, in milliseconds, after which to treat network calls to urls as failed when selecting
  * online urls.
  * @param onError - Handler to report errors thrown from eventListeners.
  */
 export interface WalletConnectConstructorArgs {
   actions: Actions
   options: WalletConnectOptions
-  defaultChainId?: number
   timeout?: number
   onError?: (error: Error) => void
 }
 
-/**
- * @param desiredChainId - The desired chainId to connect to.
- * @param preventUserPrompt - If true, will suppress user-facing interactions and only connect silently.
- */
-export interface ActivateOptions {
-  desiredChainId?: number
-  onlyIfAlreadyConnected?: boolean
-}
-
 export class WalletConnect extends Connector {
   /** {@inheritdoc Connector.provider} */
   public provider?: MockWalletConnectProvider
   public readonly events = new EventEmitter3()
 
   private readonly options: Omit<WalletConnectOptions, 'rpc'>
-  private readonly rpc: { [chainId: number]: string[] }
-  private readonly defaultChainId: number
+
+  private readonly rpcMap?: Record<number, string | string[]>
+
   private readonly timeout: number
 
-  private eagerConnection?: Promise<void>
+  private eagerConnection?: Promise<MockWalletConnectProvider>
 
-  constructor({ actions, options, defaultChainId, timeout = 5000, onError }: WalletConnectConstructorArgs) {
```

Comment:
The default chain is the first item in the options.chains array. Shall we bring back defaultChainId and manually modify options.chains so that defaultChainId is the first element in the array? Or, just ask users to organise chains array on their own.

I think I am generally fan of the latter approach, unless we care about defaultChainId being universal property across different connectors.

@zzmp?

Reply:
do all connectors respect the chains array ordering? if we wanted to make this change, i feel like it'd be nice to mark defaultChainId as deprecated and have a note that to move to chains array vs remove entirely right now, but maybe that's not worth it

Reply:
Not really, this is WalletConnect specific. I don't think the constructor arguments are enforced to be the same across different connectors, so a fair amount of customisability is allowed here. 

We could technically allow a `defaultChainId` and do not make end users aware that order of `chains` does matter (we would reorder depending on passed chain id behind the scenes), but I think that could be an unnecessary abstraction.

Reply:
got it. if it's not a convention used elsewhere then i would leave it up to your judgement - no strong opinion from me

Reply:
I think we can actually make it more explicit